src/dialects/abstract/data-types.ts

```diff
@@ -1354,6 +1354,7 @@ export class DATEONLY extends AbstractDataType<AcceptedDate> {
     }
 
     if (value) {
+      // TODO!: add a test case for #10982
```

Comment:
Here

Reply:
That has been added in https://github.com/sequelize/sequelize/pull/14505/commits/77e70f1d4b6b9c59a70dc8983fecb30c970fc3bd